### ISSUE TITLE
RestSharp error in ApiClient

### ISSUE THREAD

**Issue by EdisonHarada** [mParticle/mparticle-dotnet-sdk#3]
Hello,

I'm implementing mParticle in my API, but I'm getting an error when I try to upload the events.
I tried with both "LogEvent" and "Upload", "UploadBatchAsync", and they throw the same exception, apparently "mParticle.Client.ApiClient" is not handling something with RestSharp.

Also I'm using .NET Core 3.1 and mParticle.Sdk (1.0.0).

Exception:
```
Message: Attempt by method 'mParticle.Client.ApiClient+<ExecAsync>d__8`1<T>.MoveNext()' to access method 'RestSharp.RestClient.AddHandler(System.Func`1<RestSharp.Deserializers.IDeserializer>, System.String[])' failed.
StackTrace:    at mParticle.Client.ApiClient.<ExecAsync>d__8`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at mParticle.Api.MParticle.<UploadBatchAsync>d__23.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at MParticleService.MParticle.<SendEvents>d__4.MoveNext() in C:\API\MParticleService\MParticle.cs:line 46
```

Service Class
```
namespace MParticleService
{
    public class MParticle : IMParticle
    {
        private readonly IWebHostEnvironment _env;
        private readonly mParticle.Api.MParticle _mParticle;

        public MParticle(IWebHostEnvironment env)
        {
            _mParticle = mParticle.Api.MParticle.Start(
                new Configuration(
                    "KEY",
                    "SECRET"
                )
            );

            _env = env;
        }

        public async Task SendEvent(Event evt, UserIdentity userIdentity)
        {
            await SendEvents(new List<Event> {evt}, userIdentity);
        }

        public async Task SendEvents(IList<Event> events, UserIdentity userIdentity)
        {
            var batch = new Batch
            {
                Environment = _env.IsProduction()
                    ? BaseBatch.EnvironmentEnum.Production
                    : BaseBatch.EnvironmentEnum.Development,
                UserIdentities = userIdentity.ToBatchUserIdentity(),
                DeviceInfo = userIdentity.ToDeviceInformation()
            };

            foreach (var evt in events) batch.Events.Add(evt.ToMParticleCustomEvent());

            await _mParticle.UploadBatchAsync(batch);
        }
    }
}
```

**Comment by willpassidomo** [comment]
Hey @EdisonHarada . when you run `dotnet list package --include-transitive` what version of RestSharp do you see? 

I think you are correct in that the RestSharp transitive dependency is causing an issue, but I can't reproduce this in isolation

**Comment by willpassidomo** [comment]
fixed in version 1.0.1, should be out momentarily
